### PR TITLE
[TT-6454] modify OAS servers methods.

### PR DESCRIPTION
**Pull request by jeffy-mathew**
[changelog]
added: OAS ReplaceServers.
changed: OAS AddServers.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4373/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3343802042)

**Review by lghiur — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4373/merge
Commit: fb8a6ec724c1159f491db1a9bba345fcc8499081 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3344630666)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4373/merge
Commit: 948933292bf22b8ca45b60ef90f0db98a8a82cbd 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3347691594)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4373/merge
Commit: f7538dd04b072ca13a1bf782dd05b3d444760cf9 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3347709496)

File: apidef/oas/oas.go [hunk 1]
```diff
@@ -277,27 +277,27 @@ func (s *OAS) getTykOperations() (operations Operations) {
...
 // AddServers adds a server into the servers definition if not already present.
-func (s *OAS) AddServers(apiURL string) {
-	if len(s.Servers) == 0 {
-		s.Servers = openapi3.Servers{
-			{
-				URL: apiURL,
-			},
-		}
-		return
+func (s *OAS) AddServers(apiURLs ...string) {
```
Comment (titpetric) [inline apidef/oas/oas.go:280]: This is, technically, a breaking change - if we're doing that, could you please add an error return, and make sure to return an error when len(apiURLs) == 0 - handle the error outside of here.

On a side-note, do we need it to be public? Can it be `addServers` and left unexposed?

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 4f04f5f8b88d163f7b96823fecf9e31da5226afc [TT-6454] modify OAS servers methods. (#4373)

* modify OAS servers methods.

[changelog]
added: OAS ReplaceServers.
changed: OAS AddServers.

* retain user added servers in replace OAS servers
Triggered by: push (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3360170347)

**Pull request merged**